src/action.js

```diff
@@ -14,13 +31,17 @@ function getConfig() {
   const supportTypescript = core.getBooleanInput("support_typescript", {
     required: false,
   });
-  const githubRef = core.getInput("github_ref", { required: false });
+  const githubRef =
+    process.env.GITHUB_HEAD_REF ||
```

Comment:
command line arguments should take precedence over environment variables. That's the expected behavior from unix tools

Reply:
Yeah, this one is a little weird. As far as I can tell, there's no need to pass in `github_ref` explicitly. It's available in `process.env.GITHUB_REF`.

With that said, I'm preferring `process.env.GITHUB_HEAD_REF` to the input github_ref because github_ref for a PR is something of the form `refs/pull/9/merge`, which is almost certainly _not_ what we want. `GITHUB_HEAD_REF` is the head branch for a PR, but is not defined when merely pushing a branch.

Should I prefer the env variables, and eliminate the explicit github_ref input?

Happy to discuss via Zoom if that's easier?

Reply:
Oh in that case, make the github_ref as deprecated (just a comment is fine) or copy the above explanation into a comment so it's obvious why we are breaking convention